node/streaming_out_request.js

```diff
@@ -45,7 +47,17 @@ function StreamingOutRequest(id, options) {
     function onFrame(tup) {
         var parts = tup[0];
         var isLast = tup[1];
-        self.sendParts(parts, isLast);
+
+        onInitResponse();
+        function onInitResponse() {
+            if (self.handler.remoteHostPort !== null) {
+                self.sendParts(parts, isLast);
+            } else if (self.checkTimeout()) {
+                return;
+            } else {
+                require('timers').setImmediate(onInitResponse);
```

Comment:
you cannot loop like this; let's just making streaming be thrown exceptions when init handshake is not finished.